src/Core/Logging/Extensions/ILogServiceExtensions.Warning.cs

```diff
@@ -0,0 +1,166 @@
+﻿using System;
+using System.Runtime.CompilerServices;
+
+namespace CodeMonkeys.Logging
+{
+    public static partial class LogServiceExtensions
+    {
+        public static void Warning<TState>(
+            this ILogService service,
+            DateTimeOffset timestamp,
+            TState state,
+            Exception ex,
+            [CallerMemberName] string methodName = "")
+        {
+            service.Internal(
+                timestamp,
+                state,
+                ex,
+                methodName);
+        }
+
+        public static void Warning<TState>(
+            this ILogService service,
+            DateTimeOffset timestamp,
+            TState state,
+            [CallerMemberName] string methodName = "")
+        {
+            service.Internal(
+                timestamp,
+                state,
+                methodName);
+        }
+
+        public static void Warning(
+            this ILogService service,
+            DateTimeOffset timestamp,
+            Exception ex,
+            [CallerMemberName] string methodName = "")
+        {
+            service.Internal(
+                timestamp,
+                ex,
+                methodName);
+        }
+
+
+
+        public static void Warning<TState>(
+            this ILogService service,
+            TState message,
+            Exception ex,
+            [CallerMemberName] string methodName = "")
+        {
+            service.Internal(
+                message,
+                ex,
+                methodName);
+        }
+
+        public static void Warning<TState>(
+            this ILogService service,
+            TState message,
+            [CallerMemberName] string methodName = "")
+        {
+            service.Internal(
+                message,
+                methodName);
+        }
+
+        public static void Warning(
+            this ILogService service,
+            Exception ex,
+            [CallerMemberName] string methodName = "")
+        {
+            service.Internal(
+                ex,
+                methodName);
+        }
+
+
+
+        private static void Warning<TState>(
```

Comment:
Codacy found an issue: [Method has 5 parameters, which is greater than the 4 authorized.](https://app.codacy.com/gh/UltimateCodeMonkeys/CodeMonkeysMVVM/pullRequest?prid=5762132)